ballerina/modules/parser/lexer.bal

```diff
@@ -103,22 +103,112 @@ public class Lexer {
         string word = "";
         Location location = self.currentLocation.clone();
         _ = self.readNextChar(); // Ignore first double quote character
-        while (true) {
+        string value = self.charReader.peek();
+        if value == DOUBLE_QUOTE {
+            value = self.readNextChar();
+            value = self.charReader.peek();
+            if value == DOUBLE_QUOTE {
+                return self.readBlockStringLiteral();
+            } else {
+                return  getToken(word, T_STRING, location);
+            }
+        } else {
+            while true {
+                value = self.charReader.peek();
+                if value is LineTerminator {
+                    string message = "Syntax Error: Unterminated string.";
+                    return error UnterminatedStringError(message, line = self.currentLocation.line,
+                                                         column = self.currentLocation.column);
+                } else {
+                    value = self.readNextChar();
+                    if value == DOUBLE_QUOTE && previousChar != BACK_SLASH {
+                        break;
+                    } else {
+                        word += value;
+                    }
+                    previousChar = value;
+                }
+            }
+            return getToken(word, T_STRING, location);
+        }
+    }
+
+    isolated function readBlockStringLiteral() returns Token|SyntaxError {
+        string previousChar = "";
+        string word = "";
+        string[] lines = [];
+        Location location = self.currentLocation.clone();
+        _ = self.readNextChar(); // Ignore third double quote character
+        while true {
             string value = self.charReader.peek();
-            if (value is LineTerminator) {
-                string message = "Syntax Error: Unterminated string.";
-                return error UnterminatedStringError(message, line = self.currentLocation.line,
-                                                     column = self.currentLocation.column);
-            } else if (value == DOUBLE_QUOTE && previousChar != BACK_SLASH) {
+            if value == NEW_LINE {
+                lines.push(word);
+                word = "";
                 value = self.readNextChar();
-                break;
+            } else if value == DOUBLE_QUOTE {
+                previousChar = self.readNextChar();
+                value = self.charReader.peek();
+                if value == DOUBLE_QUOTE {
+                    previousChar = self.readNextChar();
+                    value = self.charReader.peek();
```

Comment:
Do we have a test for this?
i.e. a case where `""` occurs in a block string? If not shall we add one?

Reply:
We already have a test named ```testBlockStringsWithDoubleQuotes```.